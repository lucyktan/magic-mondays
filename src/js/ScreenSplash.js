import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Text,
  Image,
  TouchableHighlight,
  Platform
} from 'react-native';
import CheckBox from 'react-native-checkbox';
import couponImage from '../../img/coupon.png';

class ScreenSplash extends Component {

  static navigationOptions = ({ navigation }) => {
    return {
      title: `Welcome ${navigation.state.params.screen}`,
    }
  };
  constructor (prop, context) {
    super(prop, context);
    this.state = {
      isChecked: false
    };
  }

  render() {
    const { state, navigate } = this.props.navigation;
    return (
      <View style={styles.container}>
        <Text style={styles.titleText}>Happy Macy's Monday!</Text>
        <Text style={styles.basicText}>Play now to unlock your prize!</Text>
        <Image style={styles.image} source={ couponImage }/>
        <CheckBox
          label='I agree to the Terms & Conditions'
          labelStyle={styles.checkTerms}
          containerStyle={styles.checkboxContainerStyle}
          checked={this.state.isChecked}
          onChange={(checked) => this.setState({ isChecked: !checked })}
          {...console.log('Terms box checked', this.state.isChecked)}/>
          
        <View style={styles.buttonContainer}>

        { (this.state.isChecked) ? 
            <TouchableHighlight
            onPress={() => navigate("ScreenGame", {screen: "Screen Game"})} 
            style={[styles.button, {backgroundColor: '#CC0000'}]}>
            <Text style={styles.buttonText}>Play Now!</Text>
          </TouchableHighlight>
          : 
          <TouchableHighlight
            style={[styles.button, {backgroundColor: '#EB9999'}]}>
            <Text style={styles.buttonText}>Play Now!</Text>
          </TouchableHighlight>
        }

        </View>
      </View>
    );
  }
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  titleText: {
    color: '#CC0000',
    fontFamily: (Platform.OS === 'ios')? 'Helvetica Neue' : 'Roboto',
    fontSize: 30,
    paddingTop: 50,
    paddingBottom: 20,
    textAlign: 'center'
  },
  checkTerms: {
    // flexDirection: 'column',
    flexWrap: 'wrap'
  },
  checkboxContainerStyle: {
    padding: '5%',
    //width:'80%'
  },
  basicText: {
    paddingTop: 15,
    paddingBottom: 15,
    color: '#666666',
    fontSize: 20,
    textAlign: 'center'
  },
  image: {
    flex: 1,
    resizeMode: 'contain',
    paddingLeft: 20,
    paddingRight: 20
  },
  buttonContainer: {
    flex: 1,
    marginLeft: 20,
    marginRight: 20,
    marginTop: 20,
    height: 100,
    width: 100
  },
  button: {
    margin: 10,
    justifyContent: 'center',
    width: 100
  },
  buttonText: {
    color: 'white',
    alignSelf: 'center',
    fontSize: 18
  }
});
export default ScreenSplash;