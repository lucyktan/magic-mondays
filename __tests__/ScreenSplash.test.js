import {View} from 'react-native';
import React from 'react';
import { mount,configure, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ScreenSplash from '../src/js/ScreenSplash';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import { NativeModules } from 'react-native';


jest.mock('ScreenSplash', ()=> {
  return {
    actions: {
      checkbox: jest.fn(),
    },
  }
})
describe('Screen Splash', () => {
  it('Page rendered', () => {

    let navigation = {
    state: {
        key: 'id-1512512594799-2',
        params: {
	      screen: 'Screen Splash',
        user: 'prize'
  	  },
        routeName: 'ScreenSplash',
      }
    };
    const rendered = mount(
    	<ScreenSplash navigation={navigation}/>
    );
    const checkboxSpy = sinon.spy(ScreenSplash.actions, 'checkbox');
    expect(rendered.state('isChecked')).toEqual(false);
    rendered.find('CheckBox').first().simulate('valueChange');
    expect(checkboxSpy.calledOnce).toBe(true);
    

  });
});
