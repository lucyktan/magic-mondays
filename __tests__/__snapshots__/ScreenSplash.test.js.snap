// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen Splash renders correctly 1`] = `
<Image
  source={1}
  style={
    Object {
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "top": 50,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "flex": 1,
          "paddingBottom": 35,
          "resizeMode": "center",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 20,
          "fontWeight": "bold",
          "letterSpacing": 2,
          "paddingTop": 20,
          "textAlign": "center",
          "textShadowColor": "black",
          "textShadowOffset": Object {
            "height": 1,
            "width": -1,
          },
          "textShadowRadius": 10,
        }
      }
    >
      Play now to unlock your prize!
    </Text>
    <Image
      source={1}
      style={
        Object {
          "flex": 1,
          "height": 300,
          "paddingBottom": 190,
          "width": 300,
        }
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 5,
            },
            Object {
              "paddingTop": 25,
            },
          ]
        }
      >
        <Image
          source={1}
          style={
            Array [
              Object {
                "height": 26,
                "width": 26,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            Object {
              "marginLeft": 10,
              "marginRight": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "grey",
                  "fontSize": 15,
                },
                Object {
                  "color": "white",
                  "flexWrap": "wrap",
                  "fontWeight": "bold",
                  "textShadowColor": "black",
                  "textShadowOffset": Object {
                    "height": 1,
                    "width": -1,
                  },
                  "textShadowRadius": 10,
                },
              ]
            }
          >
            I agree to the Terms & Conditions
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "paddingBottom": 65,
          "paddingHorizontal": 75,
          "paddingVertical": 15,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "alignItems": "center",
                "paddingHorizontal": 65,
                "paddingVertical": 15,
              },
              Object {
                "backgroundColor": "#cbe5ef",
              },
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "white",
              "fontSize": 18,
            }
          }
        >
          Play Now!
        </Text>
      </View>
    </View>
  </View>
</Image>
`;
